bundle/Resources/public/js/views/cof-contentcreationview.js

```diff
@@ -328,6 +349,106 @@ YUI.add('cof-contentcreationview', function (Y) {
                 this.set('displayed', false);
             }
         },
+
+        /**
+         * Fetches the suggested locations.
+         *
+         * @protected
+         * @method _fetchSuggestedLocations
+         * @param event {Object} event facade
+         */
+        _fetchSuggestedLocations: function (event) {
+            this.fire('fetchSuggestedLocations', event);
+        },
+
+        /**
+         * Sets the default selected location.
+         *
+         * @protected
+         * @method _setDefaultLocation
+         * @param event {Object} event facade
+         */
+        _setDefaultLocation: function (event) {
+            var defaultLocation = event.newVal[0];
+
+            this.set('selectedLocation', {location: defaultLocation, contentInfo: defaultLocation.get('contentInfo')});
+        },
+
+        /**
+         * Render the suggested locations list.
+         *
+         * @protected
+         * @method _renderSuggestedLocations
+         * @param event {Object} event facade
+         */
+        _renderSuggestedLocations: function (event) {
+            var suggestedList = this.get('container').one(SELECTOR_SUGGESTED_LOCATIONS),
+                itemTemplate = this.get('suggestedItemTemplate'),
+                documentFragment = Y.one(document.createDocumentFragment()),
+                renderedItem;
+
+            event.newVal.forEach(function (location) {
+                var locationPath = '/' + location.get('contentInfo').get('name');
+
+                renderedItem = Y.Template.Micro.compile(itemTemplate);
+                renderedItem = renderedItem({id: location.get('id'), path: locationPath});
+
+                documentFragment.append(renderedItem);
+            });
+
+            suggestedList.setHTML(documentFragment);
+        },
+
+        /**
+         * Toggles the visibility of suggested locations list
+         *
+         * @protected
+         * @method _toggleSuggestedLocations
+         * @param event {Object} event facade
+         */
+        _toggleSuggestedLocations: function (event) {
+            var isBtnSuggested = event ? event.target.hasClass(CLASS_BUTTON_SUGGESTED) : false,
+                methodName = isBtnSuggested ? 'removeClass' : 'addClass';
+
+            this.get('container').one(SELECTOR_SUGGESTED_LOCATIONS)[methodName](CLASS_SUGGESTED_LOCATIONS_HIDDEN);
+
+            this._setClickOutsideSuggestedLocationsHandler(isBtnSuggested);
```

Comment:
This is incorrect approach. You should set the handler value here instead of calling another method.